backend/frontend_integration.py

```diff
@@ -0,0 +1,165 @@
+        sensor = db.session.query(FakeUser).join(NewSensor, NewSensor.userid == FakeUser._id).filter(NewSensor.userid == user._id).first() # sampling
+        print(sensor)
+        datasets = [Recording.from_real_data(sensor.fs, recording.ts_data) for recording in db.session.query(Recordings).filter(Recordings.sensorid == sensor.sensorid).all()]
+        print(len(datasets))
+        analysis_controller = AnalysisController(fs=sensor.fs, noise_amp=0.05)
+        metrics = analysis_controller.get_metrics(datasets)[0]._df.to_dict() #merge moment
+        print(metrics)
+        response = jsonify(metrics)
+        print(response)
+        response.headers.add('Access-Control-Allow-Origin', '*')
+        return response
+    except Exception as e:
+        return jsonify(error=f"Error processing request: {str(e)}"), HTTPStatus.BAD_REQUEST
+
+@endpoints.route('/api/get_user/', methods=['POST'])
+def get_user():
+    try:
+        data = request.get_json()
+        email = data.get('email')
+        password = data.get('password')
+
+        # get email and password from db
+        try:
+            creds = db.session.query(FakeUser).filter(FakeUser.email == email).first()
+        except:
+            print("Creds didn't work")
+
+        # authenticate
+        if creds is None:
+            return jsonify({'message': 'Invalid credentials :('}), HTTPStatus.UNAUTHORIZED
+
+        elif email != creds.email: # email is not found
+            return jsonify({'message': 'Invalid email :('}), HTTPStatus.UNAUTHORIZED
+        
+        elif email == creds.email and password == creds.password: # email and password match
+            return jsonify({'message': 'Logged in successfully'}), HTTPStatus.OK
+        
+        elif email == creds.email and password != creds.password: # email is found, but password doesn't match
+            return jsonify({'message': 'Invalid password'}), HTTPStatus.UNAUTHORIZED
+        
+        else: # shouldn't get here, but if they do they were probably unauthorized
+            return jsonify({'message': 'Invalid credentials :('}), HTTPStatus.UNAUTHORIZED
+
+    except OperationalError as e:
+        print("Operational Error :(")
+        db.session.rollback()
+        error = e
+        #return jsonify({"error": str(e)}), HTTPStatus.BAD_REQUEST
+    except Exception as e:
+        print("Exception error :(")
+        db.session.rollback()
+        error = e
+        #return jsonify({"error": str(e)}), HTTPStatus.BAD_REQUEST    
+    finally:
+        print("I'm closing!")
+        db.session.close()
+
+
+@endpoints.route('/api/create_user', methods=['POST'])
+def create_user():
+    try:
+        data = request.get_json()
+        name = data.get('name')
+        email = data.get('email')
+        password = data.get('password')
+        sensorid = data.get('sensorid')
+        print(sensorid)
+
+        timestamp = int(time.time() * 1000) 
+        userid = timestamp % 1000000 #id will be 6 digits long
```

Comment:
Wack way to create a unique id but okay

Reply:
yeah this is what I'm gonna change later probably LOL

I think the ID should populate based on sensorid, but I don't have that happening rn